File: src/main/java/ru/ezhov/rocket/action/types/TextRocketActionUi.java
package ru.ezhov.rocket.action.types;

import ru.ezhov.rocket.action.api.RocketActionConfigurationProperty;
import ru.ezhov.rocket.action.api.RocketActionSettings;
import ru.ezhov.rocket.action.notification.NotificationFactory;
import ru.ezhov.rocket.action.notification.NotificationType;

import javax.swing.JLabel;
import java.awt.Component;
import java.awt.Toolkit;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.StringSelection;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;
import java.util.Arrays;
import java.util.List;

public class TextRocketActionUi extends AbstractRocketAction {
    private static String LABEL = "label";

    @Override
    public String description() {
        return "Show text";
    }

    @Override
    public List<RocketActionConfigurationProperty> properties() {
        return Arrays.asList(createRocketActionProperty(LABEL, "Text to display", true));
    }

    @Override
    public Component create(RocketActionSettings settings) {
        String text = ConfigurationUtil.getValue(settings.settings(), LABEL);
        JLabel label = new JLabel(text);
        label.addMouseListener(new MouseAdapter() {
            @Override
            public void mouseReleased(MouseEvent e) {
                if (e.getButton() == MouseEvent.BUTTON3) {
                    Toolkit defaultToolkit = Toolkit.getDefaultToolkit();
                    Clipboard clipboard = defaultToolkit.getSystemClipboard();
                    clipboard.setContents(new StringSelection(text), null);

                    NotificationFactory.getInstance().show(NotificationType.INFO, "Text copy to clipboard");
                }
            }
        });
        return label;
    }

    @Override
    public String type() {
        return "SHOW_TEXT";
    }
}

File: src/main/kotlin/ru/ezhov/rocket/action/types/TextRocketActionUi.kt
package ru.ezhov.rocket.action.types

import ru.ezhov.rocket.action.api.RocketActionConfigurationProperty
import ru.ezhov.rocket.action.api.RocketActionSettings
import ru.ezhov.rocket.action.notification.NotificationFactory
import ru.ezhov.rocket.action.notification.NotificationType
import java.awt.Component
import java.awt.Toolkit
import java.awt.datatransfer.StringSelection
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent
import javax.swing.JLabel

class TextRocketActionUi : AbstractRocketAction() {
    override fun description(): String = "Show text"

    override fun properties(): List<RocketActionConfigurationProperty> =
            listOf(createRocketActionProperty(LABEL, "Text to display", true))

    override fun create(settings: RocketActionSettings): Component {
        val text = ConfigurationUtil.getValue(settings.settings(), LABEL)
        val label = JLabel(text)
        label.addMouseListener(object : MouseAdapter() {
            override fun mouseReleased(e: MouseEvent) {
                if (e.button == MouseEvent.BUTTON3) {
                    val defaultToolkit = Toolkit.getDefaultToolkit()
                    val clipboard = defaultToolkit.systemClipboard
                    clipboard.setContents(StringSelection(text), null)
                    NotificationFactory.notification.show(NotificationType.INFO, "Text copy to clipboard")
                }
            }
        })
        return label
    }

    override fun type(): String = "SHOW_TEXT"

    companion object {
        private const val LABEL = "label"
    }
}